bpf/unwinders/hash.h

```diff
@@ -1,58 +1,26 @@
+#include "common.h"
+
 // Avoid pulling in any other headers.
 typedef unsigned int uint32_t;
 
 // murmurhash2 from
-// https://github.com/aappleby/smhasher/blob/92cf3702fcfaadc84eb7bef59825a23e0cd84f56/src/MurmurHash2.cpp
-uint32_t MurmurHash2(const void *key, int len, uint32_t seed) {
-  /* 'm' and 'r' are mixing constants generated offline.
-     They're not really 'magic', they just happen to work well.  */
-
-  const uint32_t m = 0x5bd1e995;
-  const int r = 24;
-
-  /* Initialize the hash to a 'random' value */
+// https://github.com/aappleby/smhasher/blob/92cf3702fcfaadc84eb7bef59825a23e0cd84f56/src/MurmurHash2.cpp/*  */
 
-  uint32_t h = seed ^ len;
+unsigned long long hash_stack(stack_trace_t *stack, int seed) {
+  const unsigned long long m = 0xc6a4a7935bd1e995LLU;
+  const int r = 47;
+  unsigned long long hash = seed ^ (stack->len * m);
 
-  /* Mix 4 bytes at a time into the hash */
-
-  const unsigned char *data = (const unsigned char *)key;
-  // MAX_STACK_DEPTH * 2 = 256 (because we hash 32 bits at a time).
-  for (int i = 0; i < 256; i++) {
-    if (len < 4) {
-      break;
-    }
-    uint32_t k = *(uint32_t *)data;
+  for(int i=0; i<MAX_STACK_DEPTH; i++){
+    unsigned long long k = stack->addresses[i];
 
     k *= m;
     k ^= k >> r;
     k *= m;
 
-    h *= m;
-    h ^= k;
-
-    data += 4;
-    len -= 4;
+    hash ^= k;
+    hash *= m;
   }
 
-  /* Handle the last few bytes of the input array  */
-
-  switch (len) {
```

Comment:
We don't need the code below as we always have a stack size in bytes which is divisible by 8 bytes